WordPress/src/main/java/org/wordpress/android/ui/posts/PostListFragment.kt

```diff
@@ -264,30 +262,45 @@ class PostListFragment : Fragment() {
         scrollPosition?.let { recyclerView?.smoothScrollToPosition(it) }
     }
 
-    private fun updateEmptyViewForState(emptyViewState: PostListEmptyViewState) {
-        actionableEmptyView?.visibility = if (emptyViewState == HIDDEN_LIST) View.GONE else View.VISIBLE
-        val stringId = when (emptyViewState) {
-            HIDDEN_LIST -> return // nothing else to do!
-            EMPTY_LIST -> R.string.posts_empty_list
-            LOADING -> R.string.posts_fetching
-            REFRESH_ERROR -> if (NetworkUtils.isNetworkAvailable(nonNullActivity)) {
-                R.string.error_refresh_posts
-            } else R.string.no_network_message
-            PERMISSION_ERROR -> R.string.error_refresh_unauthorized_posts
-        }
-        val isNewPostActionVisible = emptyViewState == EMPTY_LIST
-        actionableEmptyView?.let {
-            it.image.setImageResource(R.drawable.img_illustration_posts_75dp)
-            it.image.visibility = if (isNewPostActionVisible) View.VISIBLE else View.GONE
-            it.title.setText(stringId)
-            it.button.setText(R.string.posts_empty_list_button)
-            it.button.visibility = if (isNewPostActionVisible) View.VISIBLE else View.GONE
-            it.button.setOnClickListener {
-                viewModel.newPost()
+    private fun updateEmptyViewForState(state: PostListEmptyUiState) {
+        if (state.listVisible) {
+            actionableEmptyView?.visibility = View.GONE
+        } else {
+            actionableEmptyView?.visibility = View.VISIBLE
+            actionableEmptyView?.let { view ->
+                setTextOrHide(view.title, state.titleResId)
+                setImageOrHide(view.image, state.imgResId)
+                setupButtonOrHide(view.button, state.buttonTextResId, state.onButtonClick)
             }
         }
     }
 
+    private fun setupButtonOrHide(
+        buttonView: Button,
+        textResId: Int?,
+        onButtonClick: (() -> Unit)?
+    ) {
+        buttonView.visibility = if (textResId == null) View.GONE else View.VISIBLE
```

Comment:
Nitpick: I think we can cleanup this method a little bit. Here is an approach that might work:

```
with(buttonView) {
    uiHelpers.updateVisibility(this, textResId != null)
    textResId?.let {
        setText(textResId)
        setOnClickListener { onButtonClick?.invoke() }
    }
}
```

Reply:
I think this can be done better with the `setTextOrHide` which I only just saw during the review. Here is an alternative approach:

```
setTextOrHide(buttonView, textResId)
buttonView.setOnClickListener { onButtonClick?.invoke() }
```

I haven't tried this, but I think it'll work. It should be safe to assign the `onClickListener` even if the view is not visible. It just won't do anything, because it'll never be called.

Reply:
Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/9287/commits/ea42218089634be251bd350787f301bf6a57647a